### DISCUSSION TITLE
Efficiently use ndarray::concatenate on a Vec of ArrayBase<OwnedRepr>

### DISCUSSION THREAD

**Discussion by mentics** [rust-ndarray/ndarray#1393] [Q&A]
`concatenate` expects ArrayView's (ViewRepr). I'm doing some calculations and collecting them into a Vec, so I'll have Arrays (OwnedRepr). What's the best way to concatenate in that situation?

```
let v = (0..L).map(|l| {
  a.dot(&b)
  // Can't do A.dot(B).view() here because compiler will complain returning references
}).collect::<Vec<_>>();
// how to concatenate(v)?
```

I understand I could `v.iter().map(|x| x.view()).collect::<Vec<_>>()` but it seems unfortunate and messy that I would need to create a new Vec.

**Comment by nilgoyette** [comment]
I'm not sure I understand what you're asking. I understand your title but nor your code. So I'll answer from what I understand from the title.

If you're asking for the **most beautiful way**, I don't have an answer for you. I think ndarray doesn't provide anything useful to do what you want. If I had your problem, I'd simply code a function to do it. Calculate the sum of the lengths, allocate once, then iterate and push/insert. Maye using Iter::chain with collect. can help you.

**Comment by mentics** [comment]
I apologize. I didn't write it clearly. I'm trying to use this method:
`[ndarray.concatenate](https://docs.rs/ndarray/latest/ndarray/fn.concatenate.html)`

Note that it takes a reference to a slice of `ArrayView (ArrayBase<ViewRepr)`. However, I have a Vec of `Array's (ArrayBase<OwnedRepr)`. So, I can't pass it into concatenate. It seems odd that I would need to create a new vector to convert them all to views, so I'm asking if there is some other simpler way to use that concatenate method (or equivalent) on a collection of owned Array's.

**Reply by nilgoyette** [reply]
Thank you for making your request clearer. However... sorry, no. A clean solution requires that the maintainers change the definition of concatenate (we probably won't in a timeframe that suits you). This being said, I don't know why this function only accept views. Looking at the code, I don't see anything that would forbid an owned array. It looks like it should accept both (an array of view or an array of owned, but not a mix).

I don't know what's acceptable for you, but I'd either create a new vector or create your own function that does the work.
